### PR TITLE
Setting the Healthy/Unhealhy Context for Temp Profile Dimensions RCAs

### PR DESCRIPTION
**Pull request by aditjind**
*Description of changes:* Minor Refactoring and setting the Resource context of RCAs. Context is based on a hardcoded RCA threshold. If the heat calculated along that dimension is greater than the threshold we mark the RCA as unhealthy otherwise we mark it as healthy. 

*Tests:* Manual Testing done on Docker. 

*Code coverage percentage for this patch:* Unchanged

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java [hunk 1]
```diff
@@ -88,13 +88,16 @@
 public class ElasticSearchAnalysisGraph extends AnalysisGraph {
 
   private static final Logger LOG = LogManager.getLogger(ElasticSearchAnalysisGraph.class);
+  private static final int rcaPeriod = 12;
```
Comment (rguo-aws) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java:91]: we should use snake case for static variables

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java [hunk 2]
```diff
@@ -88,13 +88,16 @@
...
+  private static final int evaluationIntervalSeconds = 5;
+
 
   @Override
   public void construct() {
-    Metric heapUsed = new Heap_Used(5);
-    Metric gcEvent = new GC_Collection_Event(5);
-    Metric heapMax = new Heap_Max(5);
-    Metric gc_Collection_Time = new GC_Collection_Time(5);
+    Metric heapUsed = new Heap_Used(evaluationIntervalSeconds);
```
Comment (rguo-aws) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java:97]: Ideally we should remove this argument in constructor and fix the value to 5 seconds in base class. Joydeep wanted us to refactor this long time ago

Comment (aditjind) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java:97]: I can see that this the evaluationInterval of the AggregateMetric is 1s. I dont know if there is any reason for that. If not, we should remove it altogether from here and fix its value to constant 5 seconds.

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java (hunk 1 above)
Comment (aditjind) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java:91]: Done

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java (hunk 2 above)
Comment (rguo-aws) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java]: let's create a separate issue to keep track of that.

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java [hunk 3]
```diff
@@ -88,13 +88,16 @@
...
   @Override
   public void construct() {
-    Metric heapUsed = new Heap_Used(5);
-    Metric gcEvent = new GC_Collection_Event(5);
-    Metric heapMax = new Heap_Max(5);
-    Metric gc_Collection_Time = new GC_Collection_Time(5);
+    Metric heapUsed = new Heap_Used(EVALUATION_INTERVAL_SECONDS);
+    Metric gcEvent = new GC_Collection_Event(EVALUATION_INTERVAL_SECONDS);
+    Metric heapMax = new Heap_Max(EVALUATION_INTERVAL_SECONDS);
+    Metric gc_Collection_Time = new GC_Collection_Time(EVALUATION_INTERVAL_SECONDS);
```
Comment (khushbr) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java:100]: Thank You for making this change.

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/temperature/dimension/CpuUtilDimensionTemperatureRca.java [hunk 4]
```diff
@@ -62,12 +64,18 @@ public void generateFlowUnitListFromWire(FlowUnitOperationArgWrapper args) {
...
     public DimensionalTemperatureFlowUnit operate() {
-        DimensionalTemperatureFlowUnit flowUnit = DimensionalTemperatureCalculator.getTemperatureForDimension(
+        DimensionalTemperatureFlowUnit cpuUtilTemperatureFlowUnit = DimensionalTemperatureCalculator.getTemperatureForDimension(
                 shardStore,
                 TemperatureVector.Dimension.CPU_Utilization,
                 CPU_UTIL_BY_SHARD,
                 AVG_CPU_UTIL_BY_SHARD, CPU_UTIL_SHARD_INDEPENDENT, CPU_UTIL_PEAK_USAGE,
                 THRESHOLD_NORMALIZED_VAL_FOR_HEAT_ZONE_ASSIGNMENT);
-        return flowUnit;
+        LOG.info("CPU Utilization temperature calculated: {}",
```
Comment (khushbr) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/temperature/dimension/CpuUtilDimensionTemperatureRca.java:73]: nit: Consider moving this to line 79 and add the context information as well.

**Pull request merged**